### PR TITLE
CES/helm: configure --ces-slice-mode via Helm values

### PR DESCRIPTION
**Pull request by thorn3r**
The supported values for the `--ces-slice-mode` option are a bit redundant (cesSliceModeIdentity, cesSliceModeFCFS). This commit shortens them to `identity` and `fcfs` (while still supporting the old variants), and adds support for configuring the CES slice mode via Helm. Some unused constants related to CiliumEndpointSlice options were also removed.

### CODE REVIEW

File: operator/pkg/ciliumendpointslice/endpointslice.go [hunk 1]
```diff
@@ -110,11 +113,16 @@ func (c *Controller) Start(ctx cell.HookContext) error {
 	c.logger.Info("Bootstrap ces controller")
 	c.context, c.contextCancel = context.WithCancel(context.Background())
 	defer utilruntime.HandleCrash()
-	if c.slicingMode == cesIdentityBasedSlicing {
+
+	switch c.slicingMode {
+	case identityMode, "cesSliceModeIdentity":
```
Comment (gandro) [inline operator/pkg/ciliumendpointslice/endpointslice.go]: nit: Can we add comment here that `cesSliceModeIdentity` and `cesSliceModeFCFS` are accepted for backwards-compatibility? 

Ideally we'd even warn the user that they are using old values and need to upgrade their configuration. That allows us to eventually remove those cases

**Review by gandro — APPROVED** [review]

**Review by pippolo84 — APPROVED** [review]
Nice, thanks! :100:

File: operator/pkg/ciliumendpointslice/endpointslice.go (hunk 1 above)
Comment (thorn3r) [reply operator/pkg/ciliumendpointslice/endpointslice.go]: Good suggestion! Added this :+1:

**Review by youngnick — APPROVED** [review]

**Review by learnitall — APPROVED** [review]
LGTM, thanks!

**Pull request merged**